src/app/_components/SortListbox.tsx

```diff
@@ -20,19 +21,15 @@ export function SortListbox({ sortId, onChange, options }: SortListboxProps) {
   const selectedOption =
     options.find((option) => option.id === sortId) || options[0]
 
+  const { isMobile } = useViewport()
```

Comment:
Ah yes, I tried `useMediaQuery` but it caused hydration errors. I guess it needs to be paired with `useIsMounted`? I find it a bit more cleaner to use 1 custom hook, versus using 2 hooks and conditions in a component? What do you think?

Reply:
Yes, you're right; it would be a bit cleaner to abstract this into one custom hook.

We could do something like this to abstract the hide/show breakpoint logic, what do you think?

```ts
import theme from 'tailwindcss/defaultTheme'
import { useMediaQuery } from 'usehooks-ts'

type BreakpointValue = keyof typeof theme.screens

export function useIsBelowBreakpoint(breakpoint: BreakpointValue) {
  const breakpointWithoutPx = theme.screens[breakpoint].replace('px', '')
  const numberBreakpoint = Number(breakpointWithoutPx)

  const isBelowBreakpoint = useMediaQuery(
    `(max-width: ${numberBreakpoint - 1}px)`,
  )

  return isBelowBreakpoint
}
```

What do you think?

Reply:
Also, we should not need to call `useIsMounted`, `useMediaQuery` already relies on [useEffect](https://github.com/juliencrn/usehooks-ts/blob/master/packages/usehooks-ts/src/useMediaQuery/useMediaQuery.ts) internally to check that the component is mounted.

The hydration error also occurs with the current `useViewport` hook. It only happens on `md` and above, where the switch between `isMobile` and `isDesktop` happens.

Let's see if the error persists after we rework the component. If it does, we use dynamic imports.

Reply:
Yes that works :)